### PR TITLE
Inital Remote Wakeup Support

### PR DESCRIPTION
**Pull request by smasher816**
Not working yet...

### CODE REVIEW

**Comment by smasher816** [comment]
So, I believe this is what needs to happen from my reading.

1) The device descriptor must specify that it supports remote wakeup, ex: bmAttributes bit 5 is set in the configuration descriptor (already set to A1). See: http://www.beyondlogic.org/usbnutshell/usb5.shtml#ConfigurationDescriptors

2) Set Feature and Clear Feature must both mark/unmark the need to resume (ex: set some variable), when the feature selector is DEVICE_REMOTE_WAKEUP (wVale=0x01). Added. See: http://www.beyondlogic.org/usbnutshell/usb6.shtml. Iirc, some forum said that the host would set this feature before suspending, then clear it after waking.

3) Get Status should probably return the remote wakeup status (bit 1). Added. See above link.

4) The usb device must signal the host that it should wakeup. Tried to add? See: USB 2.0 pdf, section 7.1.7.7. This involves some fun timing over the bus, I think (hope) setting the RESET bit of USB_CTL will do this for us?

There might be more but I believe those are the main points.

**Comment by smasher816** [comment]
TMK has support for wake from suspend. It is a different chip, but the usb feature part should be the same.

These commits might be helpful:
https://github.com/tmk/tmk_keyboard/commit/9019c901dd7b4e37006eb1f0442a57dacc0b3f0c#diff-93c0ea212c7b44aceebfef367e3cc66aL711
https://github.com/tmk/tmk_keyboard/commit/afc3148e2341e4be982cdb628543e30923cfdaf7#diff-57238d989f2bae92ae2d9891cb2ac181R61

File: Output/pjrcUSB/arm/usb_keyboard.c [hunk 1]
```diff
@@ -76,6 +76,13 @@ void usb_keyboard_send()
 	uint32_t wait_count = 0;
 	usb_packet_t *tx_packet;
 
+	if (remote_wakeup_enabled) {
+		USB0_CTL |= USB_CTL_RESUME;
+		_delay_ms(5); //wait 1 to 15ms
```
Comment (smasher816) [inline Output/pjrcUSB/arm/usb_keyboard.c]: Try increasing to atleast 10ms

**Comment by unphased** [comment]
Sorry, I don't know anything about this stuff here, but basically I have a data point for you, and it is that my Infinity keyboard does not work if I include this code. 

So, for now, I have to revert this commit back out in order to keep using my keyboard.

**Pull request merged**